### ISSUE TITLE
Is there a way to set directly the connection string?

### ISSUE THREAD

**Issue by TommyEsteban** [wagnerandrade/Thunderstruck#12]
Hey, nice job!

currently the connection string is loaded from app/web config file, but Im wondering if there is a way to set the connection string directly to the context, if not, could be it a good improvement!.

Thanks.

**Comment by GusBeare** [comment]
Yeah, see the readme:

"If you need a different default connection string, you can specify the DefaultConnectionString name to use on the DataContext class:"

DataContext.DefaultConnectionStringName = "AnotherDefaultConnection";

**Comment by TommyEsteban** [comment]
yeah, that's the name of the connection string in the configuration file, not the connection string itself, for example something like
`DataContext.ConnectionString = "database=db;uid=user;pwd=mypwd";`

**Comment by GusBeare** [comment]
why do you need to hard code the connection string there?  just create a new one in web.config and use that name.

**Comment by TommyEsteban** [comment]
because for example, we dont use app.config files, we use environment variables or command line arguments with encrypted parameters, like connection string to support multiple environments.

**Comment by GusBeare** [comment]
Ah I see.  I don't know if the original developer is still around so you might have to fork the code.

Otherwise, have you tried Simple. Data?

http://simplefx.org/simpledata/docs/pages/Start/OpeningAConnection.html

https://github.com/markrendle/Simple.Data

It's a bit out of date but I've been using it for years and love it. It definitely supports connection strings.

 e.g.

var magicDb = Database.OpenConnection("--Your Connection String--");
var artist3 = magicDb.Artists.FindByName("Iron Maiden");

There's supposed to be a .net CORE version in the pipeline but don't know about progress.

**Comment by TommyEsteban** [comment]
thanks for suggestion.

**Comment by 19WAS85** [comment]
Sorry @TommyEsteban, I haven't been around... as you can see. I'm glad if you want to provide a pull request, if you want.

Thanks @GusBeare for help and options.

**Comment by bdemarzo** [comment]
I may have a quick fix for this -- check out my fork at https://github.com/bdemarzo/Thunderstruck

Basically, I added new constructors to the DataContext class that accept a ConnectionStringSettings object. Unit tests pass and in theory it should work, but I didn't test it thoroughly.
